From 9d9e2513c09e7ba7474c94455e5f82b3d00e6396 Mon Sep 17 00:00:00 2001
From: Doug Goldstein <cardoe@cardoe.com>
Date: Mon, 2 Dec 2013 21:35:11 -0600
Subject: [PATCH] setup: Use user requested python binary

When setup.py is kicked off with a python interpreter other than the
system 'python', (e.g. python2.7 setup.py build) the build process would
switch to 'python' and not use python2.7 as requested by the user. We
should always respect the user requested python interpreter and use it.
---
 setup.py |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index bf222f8..561157c 100755
--- a/setup.py
+++ b/setup.py
@@ -122,10 +122,10 @@ class my_build(build):
     def run(self):
         apis = get_api_xml_files()
 
-        self.spawn(["python", "generator.py", "libvirt", apis[0]])
-        self.spawn(["python", "generator.py", "libvirt-qemu", apis[1]])
+        self.spawn([sys.executable, "generator.py", "libvirt", apis[0]])
+        self.spawn([sys.executable, "generator.py", "libvirt-qemu", apis[1]])
         if have_libvirt_lxc:
-            self.spawn(["python", "generator.py", "libvirt-lxc", apis[2]])
+            self.spawn([sys.executable, "generator.py", "libvirt-lxc", apis[2]])
 
         build.run(self)
 
@@ -267,7 +267,7 @@ class my_test(Command):
 
         apis = get_api_xml_files()
 
-        self.spawn(["python", "sanitytest.py", self.build_platlib, apis[0]])
+        self.spawn([sys.executable, "sanitytest.py", self.build_platlib, apis[0]])
 
 
 class my_clean(clean):
-- 
1.7.1

